nexuse2e-core/src/main/java/org/nexuse2e/ui2/rest/TransactionReportingHandler.java

```diff
@@ -17,23 +23,131 @@
+                    break;
+            }
+        }
+    }
+
+    private void getMessagesCount(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        String status = request.getParameter("status");
+        String type = request.getParameter("type");
+        long messagesCount = Engine.getInstance().getTransactionService().getMessagesCount(
+                getStatusNumberFromName(status),
+                getMessageTypeFromName(type),
+                0, 0, null, null,
+                TWO_WEEKS_AGO,
+                new Date());
+        String message = new Gson().toJson(messagesCount);
+        response.getOutputStream().print(message);
+    }
+
+    private String getStatusNumberFromName(String statusName) {
+        try {
+            return String.valueOf(MessageStatus.valueOf(statusName).getOrdinal());
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private Integer getMessageTypeFromName(String typeName) {
+        if (typeName != null) {
+            switch (typeName) {
+                case "ACKNOWLEDGEMENT":
+                    return Constants.INT_MESSAGE_TYPE_ACK;
+                case "NORMAL":
+                    return Constants.INT_MESSAGE_TYPE_NORMAL;
+                case "ERROR":
+                    return Constants.INT_MESSAGE_TYPE_ERROR;
+            }
+        }
+        return null;
+    }
+
+    private void getMessages(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        String pageIndex = request.getParameter("pageIndex");
+        String itemsPerPage = request.getParameter("itemsPerPage");
+        String status = request.getParameter("status");
+        String type = request.getParameter("type");
+        if (NumberUtils.isNumber(pageIndex) && NumberUtils.isNumber(itemsPerPage)) {
+            List<MessagePojo> reportMessages = Engine.getInstance().getTransactionService().getMessagesForReport(
+                    getStatusNumberFromName(status),
+                    0, 0, null, null,
+                    getMessageTypeFromName(type),
+                    TWO_WEEKS_AGO,
+                    new Date(),
+                    Integer.parseInt(itemsPerPage),
+                    Integer.parseInt(pageIndex),
+                    TransactionDAO.SORT_CREATED,
+                    false);
+
+            List<StatisticsMessage> messages = new LinkedList<>();
+            for (MessagePojo messagePojo : reportMessages) {
+                messages.add(new StatisticsMessage(messagePojo));
+            }
+
+            Gson gson = new GsonBuilder().registerTypeAdapter(Date.class, new DateWithTimezoneSerializer()).create();
+            String messageJson = gson.toJson(messages);
+            response.getOutputStream().print(messageJson);
+        } else {
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Required parameters: pageIndex and itemsPerPage");
```

Comment:
I think it would be good for developers to see how this endpoint is supposed to be used... so they don't send a request to "messages", get 100 results and expect this to be everything.